From d7e38b4fe9810d1823cb7f565cf7a3fc0134a343 Mon Sep 17 00:00:00 2001
From: Ankur Maheshwari <mankur@codeaurora.org>
Date: Mon, 19 Feb 2018 17:39:18 +0530
Subject: [PATCH 16/20] Camera2Client: Use StreamConfiguration to find
 available raw sizes.

Issue: ANDROID_SCALER_AVAILABLE_RAW_SIZES is deprecated, so raw stream
is configured with default width and height i.e -1.

Fix: Use ANDROID_SCALER_AVAILABLE_STREAM_CONFIGURATIONS to get available
raw sizes.

Change-Id: Iaa2c7aae12aabec0de060388c0dbe2f8311751bd
CRs-Fixed: 2188338
---
 .../api1/qticlient2/Parameters.cpp             | 18 ++++++++----------
 1 file changed, 8 insertions(+), 10 deletions(-)

diff --git a/services/camera/libcameraservice/api1/qticlient2/Parameters.cpp b/services/camera/libcameraservice/api1/qticlient2/Parameters.cpp
index 7f8dd2141..08bb0e227 100644
--- a/services/camera/libcameraservice/api1/qticlient2/Parameters.cpp
+++ b/services/camera/libcameraservice/api1/qticlient2/Parameters.cpp
@@ -3031,16 +3031,14 @@ Vector<Parameters::Size> Parameters::getAvailableJpegSizes() {
 
 Vector<Parameters::Size> Parameters::getAvailableRawSizes() {
     Vector<Parameters::Size> rawSizes;
-    const int JPEG_SIZE_ENTRY_COUNT = 2;
-    const int WIDTH_OFFSET = 0;
-    const int HEIGHT_OFFSET = 1;
-    camera_metadata_ro_entry_t availableRawSizes =
-    staticInfo(ANDROID_SCALER_AVAILABLE_RAW_SIZES);
-    for (size_t i = 0; i < availableRawSizes.count; i+= JPEG_SIZE_ENTRY_COUNT) {
-        int width = availableRawSizes.data.i32[i + WIDTH_OFFSET];
-        int height = availableRawSizes.data.i32[i + HEIGHT_OFFSET];
-        Size sz = {width, height};
-        rawSizes.add(sz);
+    Vector<StreamConfiguration> scs = getStreamConfigurations();
+    for (size_t i = 0; i < scs.size(); i++) {
+        const StreamConfiguration &sc = scs[i];
+        if (sc.isInput == ANDROID_SCALER_AVAILABLE_STREAM_CONFIGURATIONS_OUTPUT &&
+                sc.format == HAL_PIXEL_FORMAT_RAW10) {
+            Size sz = {sc.width, sc.height};
+            rawSizes.add(sz);
+        }
     }
     return rawSizes;
 }
-- 
2.17.1

